aidial_assistant/application/assistant_application.py

```diff
@@ -132,19 +147,13 @@ async def chat_completion(
             await chain.run_chat(history, callback)
         except ReasonLengthException:
             finish_reason = FinishReason.LENGTH
-        except OpenAIError as e:
-            if e.error:
-                raise HTTPException(
-                    e.error.message,
-                    status_code=e.http_status or 500,
-                    code=e.error.code,
-                )
 
-            raise
+        if callback.invocations:
+            choice.set_state(State(invocations=callback.invocations))
 
-        choice.set_state(callback.get_state())
         choice.close(finish_reason)
 
-        response.set_usage(
-            usage_publisher.prompt_tokens, usage_publisher.completion_tokens
-        )
+        response.set_usage(model.prompt_tokens, model.completion_tokens)
+
+        if discarded_messages:
```

Comment:
`if discarded_messages is not None`, otherwise we don't report `discarded_messages=0`.